semanticz-connector-neptune/src/main/java/zone/cogni/semanticz/connectors/neptune/NeptuneSparqlService.java

```diff
@@ -0,0 +1,131 @@
+            upload(model, null); // Upload to the default graph
+        } catch (IOException e) {
+            log.error("Error reading TTL file", e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public Model executeConstructQuery(String query) {
+        Query parsedQuery = QueryFactory.create(query, Syntax.syntaxARQ);
+        try (RDFConnection conn = getConnection();
+             QueryExecution qExec = conn.query(parsedQuery)) {
+            return qExec.execConstruct();
+        } catch (Exception e) {
+            log.error("Error executing CONSTRUCT query on Neptune", e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void executeUpdateQuery(String updateQuery) {
+        try (RDFConnection conn = getConnection()) {
+            log.debug("Executing update query:\n{}", updateQuery);
+            conn.update(updateQuery);
+        } catch (Exception e) {
+            log.error("Error executing UPDATE query on Neptune. Query:\n{}", updateQuery, e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public boolean executeAskQuery(String query) {
+        Query parsedQuery = QueryFactory.create(query, Syntax.syntaxARQ);
+        try (RDFConnection conn = getConnection();
+             QueryExecution qExec = conn.query(parsedQuery)) {
+            return qExec.execAsk();
+        } catch (Exception e) {
+            log.error("Error executing ASK query on Neptune", e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void upload(Model model, String graphUri) {
+        try (RDFConnection conn = getConnection()) {
+            if (graphUri == null || graphUri.isEmpty()) {
+                conn.load(model);
+            } else {
+                conn.load(graphUri, model);
+            }
+        } catch (Exception e) {
+            log.error("Error uploading model to graph {} in Neptune", graphUri, e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public <R> R executeSelectQuery(String query, Function<ResultSet, R> resultHandler) {
+        Query parsedQuery = QueryFactory.create(query, Syntax.syntaxARQ);
+        try (RDFConnection conn = getConnection();
+             QueryExecution qExec = conn.query(parsedQuery)) {
+            ResultSet resultSet = qExec.execSelect();
+            return resultHandler.apply(resultSet);
+        } catch (Exception e) {
+            log.error("Error executing SELECT query on Neptune", e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void dropGraph(String graphUri) {
+        String deleteQuery = String.format("DROP GRAPH <%s>", graphUri);
+        executeUpdateQuery(deleteQuery);
+    }
+
+    @Override
+    public void updateGraph(String graphUri, Model model) {
+        upload(model, graphUri);
+    }
+
+    @Override
+    public void replaceGraph(String graphUri, Model model) {
```

Comment:
strange. is this not the default implementation in the interface??

Reply:
yes, should be default implementaton